// Copyright (c) farm-ng, inc.
//
// Licensed under the Amiga Development Kit License (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     https://github.com/farm-ng/amiga-dev-kit/blob/main/LICENSE
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package farm_ng.api.proto;

enum ServiceState {
  UNKNOWN = 0;
  STOPPED = 1;
  // Running, and actively serving data
  RUNNING = 2;
  // Idle, waiting for clients
  IDLE = 3;
  // Service isn't running at all
  UNAVAILABLE = 4;
  // Service is in some error state
  ERROR = 5;
}

message GetServiceStateRequest {
}

message GetServiceStateReply {
  ServiceState state = 1;
  int pid = 2;
  double uptime = 3;
}

service Service {
  rpc getServiceState(GetServiceStateRequest) returns (GetServiceStateReply) {}
}
